### ISSUE TITLE
Got error: "Running the contributed command:'ccls.showReferences' failed." when clicking  ref links

### ISSUE THREAD

**Issue by Will-GXZ** [MaskRay/ccls#153]
When I click ref links, I got the following error message: "Running the contributed command:'ccls.showReferences' failed."

And following log in console: 

```javascript
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:294 [Extension Host] [UriError]: Scheme is missing: {scheme: "", authority: "", path: "null", query: "", fragment: ""}
t.log @ /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:294
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:2429 Error: Running the contributed command:'ccls.showReferences' failed.
	at e._executeContributedCommand (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:460:256)
	at e.$executeContributedCommand (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:460:648)
	at t._doInvokeHandler (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:609:757)
	at t._invokeHandler (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:609:423)
	at t._receiveRequest (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:608:15)
	at t._receiveOneMessage (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:606:957)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:604:773
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:103:886
	at e.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:105:344)
	at a (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:156:881)
	at Socket._socketDataListener (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:157:95)
	at emitOne (events.js:116:13)
	at Socket.emit (events.js:211:7)
	at addChunk (_stream_readable.js:263:12)
	at readableAddChunk (_stream_readable.js:250:11)
	at Socket.Readable.push (_stream_readable.js:208:10)
	at Pipe.onread (net.js:594:20)
t.onDidNotificationChange @ /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:2429
```
-------
### Environment:
```
OS: MacOS 10.14.1 (18B75) majave
vscode: 1.29.1
vscode-ccls: 0.1.18
```
-----
I saw the same issue before [link](https://github.com/MaskRay/vscode-ccls/issues/1), but on my side it seems that the problem is still there.

**Comment by Will-GXZ** [comment]
> Can you write a log statement around `vscode-ccls/src/extension.ts:480`?
> 
> ```
> commands.registerCommand(
>     'ccls.showReferences',
>     (uri: string, position: ls.Position, locations: ls.Location[]) => {
>       commands.executeCommand(
>           'editor.action.showReferences', p2c.asUri(uri),
>           p2c.asPosition(position), locations.map(p2c.asLocation));
>     });
> ```

Sure, I added a log statement as the following: 
```javascript
    commands.registerCommand(
        'ccls.showReferences',
        (uri: string, position: ls.Position, locations: ls.Location[]) => {
          console.log("uri: " + uri + ", position: " + position + ", location: " + locations);
          commands.executeCommand(
              'editor.action.showReferences', p2c.asUri(uri),
              p2c.asPosition(position), locations.map(p2c.asLocation));
        });
```
<br/>
And I got the following message in console:

```
Extension Host
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:3267 Debugger attached.

/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:303 [Extension Host] Congratulations, your extension "code-gnu-global" is now active! (at activate (/Users/Guo/.vscode/extensions/austin.code-gnu-global-0.2.2/out/src/extension.js:15:13))
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:303 [Extension Host] Starting ccls in undefined (at getLanguageClient (/Users/Guo/Desktop/vscode-ccls/out/extension.js:243:21))
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:303 [Extension Host] Extension 'ccls-project.ccls' uses a document selector without scheme. Learn more about this: https://go.microsoft.com/fwlink/?linkid=872305 (at e (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:732:772))
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:303 [Extension Host] [ccls-project.ccls] Accessing a resource scoped configuration without providing a resource is not expected. To get the effective value for 'editor.codeLens', provide the URI of a resource or 'null' for any resource. (at e._validateConfigurationAccess (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:418:449))
t.log @ /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:303
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:303 [Extension Host] uri: null, position: undefined, location: undefined (at vscode_1.commands.registerCommand (/Users/Guo/Desktop/vscode-ccls/out/extension.js:394:25))
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:303 [Extension Host] [UriError]: Scheme is missing: {scheme: "", authority: "", path: "null", query: "", fragment: ""} (at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:76:295)
t.log @ /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:303
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:2509 Error: Running the contributed command:'ccls.showReferences' failed.
	at e._executeContributedCommand (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:501:551)
	at e.$executeContributedCommand (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:502:6)
	at t._doInvokeHandler (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:658:837)
	at t._invokeHandler (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:658:509)
	at t._receiveRequest (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:657:126)
	at t._receiveOneMessage (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:656:59)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:653:895
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:130:793
	at e.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:133:218)
	at a (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:186:147)
	at Socket.n._socketDataListener (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:186:368)
	at emitOne (events.js:116:13)
	at Socket.emit (events.js:211:7)
	at addChunk (_stream_readable.js:263:12)
	at readableAddChunk (_stream_readable.js:250:11)
	at Socket.Readable.push (_stream_readable.js:208:10)
	at Pipe.onread (net.js:594:20)
t.onDidNotificationChange @ /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:2509
```

The logged message from that line is: `uri: null, position: undefined, location: undefined`

**Comment by MaskRay** [comment]
There was an similar issue about "ccls.showReferences" which has been fixed by https://github.com/MaskRay/vscode-ccls/pull/13/files You might be using an older version

I'm closing this as it cannot be reproduced. You may attach more information (you can do that by editing your first comment), in particular, a minimal reproduce if you think this is still an issue.

**Comment by Will-GXZ** [comment]
Yes, you were right, I was using an older version which is available from
vscode extension market. Now I switched to the local build from the latest
version in github. So it should be fine. Thanks for your work, it's really
helpful.

**Comment by MaskRay** [comment]
Published 1.19 https://marketplace.visualstudio.com/items?itemName=ccls-project.ccls

**Comment by Will-GXZ** [comment]
Thanks, I just updated it. Now I found another issue that the ccls server
is not terminated after I exit vscode. I have created a new issue here
https://github.com/MaskRay/ccls/issues/209 . I'm not sure if this is
because of my local configuration problem. If you need more information
please let me know.
